packages/main/src/plugin/message-box.ts

```diff
@@ -31,14 +31,26 @@ export interface MessageBoxOptions {
    * The primary message.
    */
   message: string;
+  /**
+   * An additional (optional) detailed message.
+   */
+  detail?: string;
   /**
    * Text for buttons.
    */
-  buttons: string[];
+  buttons?: string[];
   /**
    * The (optional) type, one of 'none' | 'info' | 'error' | 'question' | 'warning'.
    */
   type?: string;
+  /**
+   * The (optional) index of the default button.
+   */
+  defaultId?: number;
```

Comment:
As this is an index why it's not being named defaultIndex ? and cancelIndex ?

Reply:
I should have included a link to the Electron messagebox API: https://www.electronjs.org/docs/latest/api/dialog#dialogshowmessageboxbrowserwindow-options.

I have intentionally been keeping a (subset of the) same API to make it easy to port - e.g. if you look at the full source of index.ts there are a few places already using cancelId. We could create our own API, I just felt simple porting was more important in this case.